### PR TITLE
[EPMEDU-1269]: Clean Architecture Adjustments

### PR DESCRIPTION
**Pull request by nuriyevn**
Refactoring improvement list:
1. Rename AuthAPI to AuthApiImpl and replace its usage with a new AuthApi interface.
2. All repository interfaces should be moved to domain layer.
Rationale: Since domain layer does not have to know about data or presentation layer.

### CODE REVIEW

**Review by VEZE — APPROVED** [review]

File: library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt [hunk 1]
```diff
@@ -1,152 +1,37 @@
...
-
-    fun setFacebookAuthenticationType(isPhoneNumberVerified: Boolean) {
-        authenticationType = AuthenticationType.Facebook(isPhoneNumberVerified)
-    }
+    var authenticationType: AuthenticationType
 
+    suspend fun getCurrentUserId(): String
+    suspend fun isSignedIn(): Boolean
+    fun setMobileAuthenticationType()
+    fun setFacebookAuthenticationType(isPhoneNumberVerified: Boolean)
```
Comment (Fileek) [inline library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt]: I guess these two functions are redundant now. Aren't they?

**Review by Fileek — COMMENTED** [review]
Please merge latest changes from develop branch and take a look at my comment

File: library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt (hunk 1 above)
Comment (nuriyevn) [reply library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt]: correct, I removed them as well. Thanks for pointing out.

File: library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt [hunk 2]
```diff
@@ -1,178 +1,35 @@
...
-import com.amplifyframework.auth.cognito.options.AWSCognitoAuthSignInOptions
-import com.amplifyframework.auth.cognito.options.AuthFlowType
-import com.amplifyframework.auth.exceptions.SessionExpiredException
-import com.amplifyframework.auth.options.AuthSignUpOptions
-import com.amplifyframework.auth.result.AuthSessionResult
-import com.amplifyframework.core.Amplify
-import com.epmedu.animeal.extensions.suspendCancellableCoroutine
-import kotlin.coroutines.resume
-import kotlin.coroutines.resumeWithException
+@Suppress("ComplexInterface")
```
Comment (Fileek) [inline library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt]: Can you try to remove this?

File: library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import com.amplifyframework.auth.cognito.options.AuthFlowType
+import com.amplifyframework.auth.exceptions.SessionExpiredException
+import com.amplifyframework.auth.options.AuthSignUpOptions
+import com.amplifyframework.auth.result.AuthSessionResult
+import com.amplifyframework.core.Amplify
+import com.epmedu.animeal.extensions.suspendCancellableCoroutine
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
+
+@Suppress("TooManyFunctions")
```
Comment (Fileek) [inline library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt]: check also this one please

File: library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt (hunk 2 above)
Comment (nuriyevn) [reply library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPI.kt]: removed, works

File: library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt (hunk 3 above)
Comment (nuriyevn) [reply library/auth/src/main/java/com/epmedu/animeal/auth/AuthAPIImpl.kt]: removed, works

**Review by Fileek — APPROVED** [review]

**Pull request merged**